packages/router/src/utils/collection.ts

```diff
@@ -45,8 +45,19 @@ export function shallowEqual(a: Params, b: Params): boolean {
  */
 export function equalArraysOrString(a: string|string[], b: string|string[]) {
   if (Array.isArray(a) && Array.isArray(b)) {
-    if (a.length != b.length) return false;
-    return a.every(aItem => b.indexOf(aItem) > -1);
+    if (a.length !== b.length) return false;
+    // For each value in a, replace the corresponding value in b
+    // with null so that it is not double counted.
+    const bCopy: (string|null)[] = [...b];
+    return a.every((val: string, index: number) => {
+      const correspondingIndex: number = bCopy.indexOf(a[index]);
+      if (correspondingIndex === -1) {
+        return false;
+      } else {
+        bCopy[correspondingIndex] = null;
+        return true;
+      }
+    });
```

Comment:
```suggestion
    if (a.length !== b.length) return false;
    const bSorted = [...b].sort();
    const aSorted = [...a].sort();
    return aSorted.every((val, index) => Object.is(bSorted[index], val));
```
How about sorting both arrays and then just comparing index by index?